### PR TITLE
chore: update deepspeed to 0.8.3 [MLG-399]

### PR DESCRIPTION
**Pull request by tayritenour**
## Description

Now that we have successfully built the `deepspeed==0.8.3` environment images, actually use them.

I confirmed all deepspeed examples that already worked continue to work with the new version.

Some of the environments images had `tf-2.8` in it, and this doesn't seem to have any particular impact on our examples.

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

**Comment by garrett361** [comment]
> Some of the environments images had tf-2.8 in it, and this doesn't seem to have any particular impact on our examples.

Yeah, DeepSpeed is built on top of pytorch, so it shouldn't have any tensorflow dependencies at all

**Review by garrett361 — APPROVED** [review]
LGTM!
